File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int n=nums.size();
        vector<int>v;
        v.push_back(-1);
        v.push_back(-1);
        int start=0;
        int end=n-1;
        int first=-1;
        while(start<=end){
            int mid=start+(end-start)/2;
            if (target==nums[mid]){
                first=mid;
                end=mid-1;
            }
            else if (target < nums[mid]){
                end=mid-1;
            }
            else{
                start=mid+1;
            }
        }
        int second=-1;
        start=0;
        end=n-1;
        while(start<=end){
            int mid=start+(end-start)/2;
            if (target==nums[mid]){
                second=mid;
                start=mid+1;
            }
            else if (target < nums[mid]){
                end=mid-1;
            }
            else{
                start=mid+1;
            }
        }
        if (first!=-1){
            v[0]=first;
        }
        if (second!=-1){
            v[1]=second;
        }
        return v;
    }
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int res1=-1;
        int res2=-1;
        int lo=0;
        int hi=nums.length-1;
        int[] arr = new int[2];
        while(lo<=hi){
            int mid=lo+(hi-lo)/2;
            if (nums[mid]==target){
                res1=mid;
                lo=mid+1;
            }
            else if (nums[mid]<target){
                lo=mid+1;
            }
            else{
                hi=mid-1;
            }
        }
        lo=0;
        hi=nums.length-1;
        while(lo<=hi){
            int mid=lo+(hi-lo)/2;
            if (nums[mid]==target){
                res2=mid;
                hi=mid-1;
            }
            else if (nums[mid]<target){
                lo=mid+1;
            }
            else{
                hi=mid-1;
            }
        }
        arr[0]=res2;
        arr[1]=res1;
        return arr;
    }
}